docs/index.adoc

```diff
@@ -94,15 +99,11 @@ class CheckSum implements Callable<Integer> {
 }
 ----
 
-Implement `Runnable` or `Callable`, and your command can be <<Executing Commands,executed>> in one line of code.
-The example above uses the `CommandLine.execute` method
-to parse the command line, handle errors, handle requests for usage and version help, and invoke the business logic.
-Applications can call `System.exit` with the returned exit code to signal success or failure to their caller.
-
-The <<Mixin Standard Help Options,mixinStandardHelpOptions>> attribute adds `--help` and `--version` options to your application.
+_This is how it works_: Create a class ➊ that represents your command and annotate this class with @Command ➑. Let this class implement `Runnable` or `Callable` and picocli will kick off your application after parsing is successfully completed. Then, annotate the fields or methods of the command class with @Parameters ➋ and/or @Option ➌ to declare what positional parameters and/or options your application expects. Picocli will convert the command line arguments given to your app to strongly typed values in the fields ➍ of your class. In the `main` method of your class, use the `CommandLine.execute` method ➎ to parse the command line, handle errors, handle requests for usage and version help, and invoke the business logic ➏. Once your business logic finished, you may call `System.exit` ➐ to return an exit code to your caller, indicating success or failure of your business logic. Additionally, you may make use of the <<Mixin Standard Help Options,mixinStandardHelpOptions>> attribute ➑  which adds `--help` and `--version` options to your application.
 
+TIP: Picocli also provides a <<Programmatic API,programmatic API>>, separately from the annotations API.
```

Comment:
I am in two minds about this change.
It is more detailed, and in a way, the example now has become something of a mini-tutorial by itself.

This is useful, but perhaps not at this place in the manual: in the introduction we still want to _sell_ the product. At this stage, readers are not yet convinced that they want to use picocli.

I would like casual readers to be able to glance at the example and think to themselves: "that looks easy! I can definitely do that! Wow, that is like a full app in 25 lines of simple code, nice!"  👍 

It is true that there is a lot of detail to be explained, but we have the whole rest of the manual to do that... 😄

Reply:
> I am in two minds about this change.
> It is more detailed, and in a way, the example now has become something of a mini-tutorial by itself.

I see the point.

> This is useful, but perhaps not at this place in the manual: in the introduction we still want to _sell_ the product. At this stage, readers are not yet convinced that they want to use picocli.

O.k. What about leaving the introductory sample as is and moving the 'mini tutorial' to the quick guide (chapter 2.1 Example). We can then place a link in the manual: "For a detailed explanation of the sample application, you may refer to the <<link,quick guide>>. 

> It is true that there is a lot of detail to be explained, but we have the whole rest of the manual to do that... smile

And additionally, we do have the quick guide ;-)

If you agree to that, I'm willing to modify my PR in the sense described above.

Reply:
Yes, that's a great idea!

The Quick Guide is currently just a shorter version of the full manual, but I like the idea of developing it into something that stands on its own more. 

This would be a nice step in that direction.